Sources/SPTDataLoaderResponse.m

```diff
@@ -62,8 +63,19 @@ - (instancetype)initWithRequest:(SPTDataLoaderRequest *)request response:(nullab
                                              code:httpResponse.statusCode
                                          userInfo:nil];
             }
-            _responseHeaders = httpResponse.allHeaderFields;
+            NSString* value = [_responseHeaders valueForKey:SPTDataLoaderURLCacheSpecialHeader];
```

Comment:
Did you mean to use `httpResponse.allHeaderFields` instead of `_responseHeaders` here?
Otherwise it looks to me you are using an uninitialised member

Reply:
good catch. thank you. working on tests now.